app/javascript/packages/captcha-submit-button/captcha-submit-button-element.ts

```diff
@@ -0,0 +1,70 @@
+class CaptchaSubmitButtonElement extends HTMLElement {
+  connectedCallback() {
+    this.button.addEventListener('click', (event) => this.handleButtonClick(event));
+  }
+
+  get isExempt(): boolean {
+    return this.hasAttribute('exempt');
```

Comment:
would it make sense to export an EXCEMPT_ATTR constant and import it in the phone input elem? to minimize typos across files?

Reply:
> would it make sense to export an EXCEMPT_ATTR constant and import it in the phone input elem? to minimize typos across files?

In earlier iterations I made this a property of the class, so that an external reference could assign a boolean value and take advantage of class typings (the latter addressing your concern, I believe).

```ts
submitButton.exempt = true;
```

I changed direction because I was seeing some strange behavior with race conditions of when each component initialized, though in retrospect I think it'd be odd that the initialization would clear any assigned instance members? It might also not be strictly relevant anymore since I also split out handling the feature-disabled case.

(For context, the original problem was that I needed the phone input to set U.S./feature-disabled as exempt immediately on page load, but PhoneInputElement would try to assign this during its initialization before CaptchaSubmitButtonElement was initialized)

I'll see if I can try to get that original approach working, since that would be a nice programmatic interface to have.

Reply:
I put together a CodePen demonstrating the problem I was dealing with:

https://codepen.io/aduth/pen/wvxbmdM

>I was seeing some strange behavior with race conditions of when each component initialized, though in retrospect I think it'd be odd that the initialization would clear any assigned instance members?

As seen in the demo above, I think the "clearing" I was seeing is due to the class property initializer.

i.e. Something like this:

```ts
class CaptchaSubmitButtonElement extends HTMLElement {
  exempt: boolean;
}
```

...is equivalent to:

```ts
class CaptchaSubmitButtonElement extends HTMLElement {
  constructor() {
    this.exempt = undefined;
  }
}
```

... therefore, it _would_ be expected that any value assigned to `exempt` prior to the component's initialization would be overridden.

Reply:
I updated the [CodePen from the previous comment](https://codepen.io/aduth/pen/wvxbmdM) to add a new `AduthComponentFiveElement` which could work as a compatible implementation to follow. It also leverages [attribute reflection](https://web.dev/custom-elements-best-practices/#:~:text=Aim%20to%20keep%20primitive%20data%20attributes%20and%20properties%20in%20sync%2C%20reflecting%20from%20property%20to%20attribute%2C%20and%20vice%20versa.), which can be a "good practice".

Reply:
I considered another idea where we set the captcha token hidden input as `required` based on whether we expect that the user must complete the challenge, which would be a nice way to take advantage of an existing property of an input element.

e.g.

```ts
captchaTokenInput.required = !isExempt;
```

I'm not sure I like the idea of drilling into the custom element's children to assign an attribute, though, vs. interfacing only with either the root custom element (`<lg-captcha-submit-button />`) or its JavaScript interface.

Reply:
Next idea: The reCAPTCHA button dispatches a custom event indicating that it plans to invoke a challenge, but will consider [`defaultPrevented`](https://developer.mozilla.org/en-US/docs/Web/API/Event/defaultPrevented) to skip the challenge. `PhoneInputElement` can listen for this event and prevent its default behavior iff the currently selected country would be exempt. Nice thing about this is it changes dependency from the time of initialization to the time that the challenge is invoked, without making the captcha implementation aware of the phone input.

Reply:
Implemented the event approach in 582c38c.